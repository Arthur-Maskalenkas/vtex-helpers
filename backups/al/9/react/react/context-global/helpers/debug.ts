import { Quadrants } from "../_interfaces";

const quadrantToValidate: Quadrants = 'quadrantTopRight'

export const debugForSpecificQuadrant = (currentQuadrant: Quadrants) => quadrantToValidate === currentQuadrant
