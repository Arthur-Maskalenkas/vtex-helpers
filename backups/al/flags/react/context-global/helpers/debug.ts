export const debugLog = (...logMessage: any) => {
	let isDebug = true

	if (isDebug) {
		console.log(...logMessage)
	}
}