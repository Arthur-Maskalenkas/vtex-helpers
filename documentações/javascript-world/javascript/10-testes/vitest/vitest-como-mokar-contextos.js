import { beforeEach, describe, expect, it, SpyInstance, vi } from "vitest";
import { Header } from "../../../../../src/views/screens/searchProduct/components/header.tsx";
import { render } from "../../../../utils/test-utils.tsx";
import * as SearchProductContextModule from "../../../../../src/views/screens/searchProduct/context/index.tsx";

describe(`${Header.name} Tests Suite`, () => {

  let searchProductContextModule: SpyInstance<any, any>

  beforeEach(() => {
    searchProductContextModule = vi.spyOn(SearchProductContextModule, 'useSearchProductContext')
  })

  it('Should render correctly2', () => {
    searchProductContextModule.mockImplementation(() => {
      return {
        states: {
          currentModule: 'product'
        } as any,
        dispatch: vi.fn()
      }
    })


    const { getByRole } = render(<Header />);


    expect(getByRole('heading', { name: 'product' })).toBeInTheDocument()
  })
})

//
//
// OU
//
//
import * as ModuleSearchProductContext from "../../../../../../../src/views/screens/searchProduct/context";
import { useSearchProductContext } from "../../../../../../../src/views/screens/searchProduct/context";

describe(() => {
  let params, searchProductContextSpy

  beforeEach(() => {
    searchProductContextSpy = vi.spyOn(ModuleSearchProductContext, 'useSearchProductContext').mockReturnValue({
      dispatch: vi.fn(),
      states: {} as any
    })
  })

  it('Should call dispatch ACTION_HANDLE_CHANGE_TO_MODULE_PRODUCT when click on button "Ver mais informações"', () => {
    const { getByRole } = render(<ProductSummary {...params} />)

    const button = getByRole('button', { name: /ver mais informações/i })

    userEvent.click(button)

    const contextSpy = useSearchProductContext()

    expect(contextSpy.dispatch).toHaveBeenCalledWith({
      type: 'ACTION_HANDLE_CHANGE_TO_MODULE_PRODUCT',
      payload: { data: params.product }
    })

  })


})