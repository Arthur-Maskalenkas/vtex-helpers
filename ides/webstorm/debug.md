# debug
caminho de configs: ctrl + shift + s && *Preferências | Construção, Execução, Implantação | Depurador | Etapas* ).

[https://www.jetbrains.com/help/webstorm/stepping-through-the-program.html](https://www.jetbrains.com/help/webstorm/stepping-through-the-program.html)

# botões
A ação *Step over (F8)* moverá a execução no arquivo atual, linha por linha, sem entrar em nenhuma chamada de função.

Com *Step Into (F7)* , passaremos para a próxima linha executada neste ou em outro arquivo. Portanto, se for uma chamada de função, ela pulará para sua definição e parará na primeira linha.

*Step Out (Shift-F8)* terminará a execução da função atual e parará na próxima instrução após a chamada.

f9 vai para a proxima linha