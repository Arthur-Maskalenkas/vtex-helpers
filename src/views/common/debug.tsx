export const Debug = ({ data }: { data: Record<any, any> }) => {
		const renderData = (data: Record<any, any>) => {
				return (
						<ul>
								{Object.entries(data).map(([ key, value ], index) => {
										if (value === null || value === undefined) {
												return (
														<li key={index}>
																<h1>{key}</h1>
																<p>{value === null ? 'null' : 'undefined'}</p>
														</li>
												)
										}

										return (
												<li key={index}>
														<h1>{key}</h1>
														{typeof value === 'object' ? renderData(value) : <p>{value}</p>}
												</li>
										)
								})}
						</ul>
				);
		};

		return <div>{renderData(data)}</div>;
}