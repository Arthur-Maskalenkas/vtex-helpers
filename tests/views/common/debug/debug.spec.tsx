import { describe, expect, it } from "vitest";
import { render } from "../../../utils/test-utils.tsx";
import { faker } from "@faker-js/faker";
import { Debug } from "../../../../src/views/common/debug.tsx";



describe(`${Debug.name} Tests Suite`, () => {

		it('Should render object', () => {
				const param = {
						name: faker.lorem.word()
				}

				const { getByText } = render(
						<Debug data={param}/>
				)

				expect(getByText(param.name)).toBeInTheDocument()
		})

		it('Should render object with nested object', () => {
				const param = {
						name: faker.lorem.word(),
						nested1: {
								name1: faker.lorem.word(),
								nested2: {
										name2: faker.lorem.word(),
										nested3: {
												name3: faker.lorem.word()
										}
								}
						}
				}

				const { getByText } = render(
						<Debug data={param}/>
				)

				expect(getByText('nested1')).toBeInTheDocument()
				expect(getByText('nested2')).toBeInTheDocument()
				expect(getByText('nested3')).toBeInTheDocument()

				expect(getByText('name1')).toBeInTheDocument()
				expect(getByText('name2')).toBeInTheDocument()
				expect(getByText('name3')).toBeInTheDocument()

				expect(getByText(param.name)).toBeInTheDocument()
				expect(getByText(param.nested1.name1)).toBeInTheDocument()
				expect(getByText(param.nested1.nested2.name2)).toBeInTheDocument()
				expect(getByText(param.nested1.nested2.nested3.name3)).toBeInTheDocument()
		})

		it("Should render object with nested array", () => {
				const params = {
						nested: [
								{
										nested: [
												{
														name: 'nested element level 2, index 0'
												},
												{
														nested: [
																{
																		name: 'nested element level 3, index 0'
																},

																{
																		name: 'nested element level 3, index 1'
																}
														]
												}
										]
								}
						]
				}

				const { getByText } = render(
						<Debug data={params}/>
				)

				expect(getByText('nested element level 2, index 0')).toBeInTheDocument()
				expect(getByText('nested element level 3, index 0')).toBeInTheDocument()
				expect(getByText('nested element level 3, index 1')).toBeInTheDocument()
		})

		it('SHould render objects,nested	objects and nested arrays', () => {
				const params = {
						nameNormalObject: 'any_value',
						nestedObject: {
								nameNestedObj: 'any_value',
						},
						nestedArray: [
								{
										nameNestedArray: 'any_value',
								}
						]
				}
				const { container } = render(
						<Debug data={params}/>
				)

				expect(container).toMatchSnapshot()

		})

		it('Should prevent null	or undefined', () => {
				const params = {
						nameNormalObject: 'any_value',
						attrNull: null,
						attrUndefined: undefined,
				}

				const { container } = render(
						<Debug data={params}/>
				)

				expect(container).toMatchSnapshot()
		})
})