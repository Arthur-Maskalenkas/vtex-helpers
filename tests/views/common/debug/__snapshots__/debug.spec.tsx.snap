// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Debug Tests Suite > SHould render objects,nested	objects and nested arrays 1`] = `
<div>
  <div>
    <ul>
      <li>
        <h1>
          nameNormalObject
        </h1>
        <p>
          any_value
        </p>
      </li>
      <li>
        <h1>
          nestedObject
        </h1>
        <ul>
          <li>
            <h1>
              nameNestedObj
            </h1>
            <p>
              any_value
            </p>
          </li>
        </ul>
      </li>
      <li>
        <h1>
          nestedArray
        </h1>
        <ul>
          <li>
            <h1>
              0
            </h1>
            <ul>
              <li>
                <h1>
                  nameNestedArray
                </h1>
                <p>
                  any_value
                </p>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Debug Tests Suite > Should prevent null	or undefined 1`] = `
<div>
  <div>
    <ul>
      <li>
        <h1>
          nameNormalObject
        </h1>
        <p>
          any_value
        </p>
      </li>
      <li>
        <h1>
          attrNull
        </h1>
        <p>
          null
        </p>
      </li>
      <li>
        <h1>
          attrUndefined
        </h1>
        <p>
          undefined
        </p>
      </li>
    </ul>
  </div>
</div>
`;
